visualization/app/codeCharta/codeCharta.model.ts

```diff
@@ -55,7 +55,7 @@ export interface CodeMapNode {
 	}
 	link?: string
 	path?: string
-	visible?: boolean
+	isBlacklisted: BlacklistType
```

Comment:
Why is this property required? Is it really necessary to add the property to all mock data etc only to fulfill the interface?

Reply:
While this can be undefined, it's not required.
If you plan on separating it into isExcluded and isFlattened we should make it required. Otherwise we'd get a lot of errors in our application. 

In the future we should work out an interface ExportCodeMapNode that does not have these properties when we import or export a file.

Reply:
> Otherwise we'd get a lot of errors in our application

@NearW I am still not understanding why we can not make those two properties optional. We could save so many lines (that you added everywhere in the node object). Those are the ones you mean with this sentence, don't you?

Reply:
Because those properties are required for a lot of calculations. Making them required will prevent us from unexpected errors, don't you think?
I don't think that adding new lines to some test-data is a problem at all.

Reply:
I was having a look at the usages and would guess that we can decorate those properties within the method `decorateIfPossible()` in the `preRender.service`

Reply:
Good idea. Done